cpp/src/community/k_truss_impl.cuh

```diff
@@ -41,8 +42,175 @@
+  auto const minor_comm_size = minor_comm.get_size();
+
+  auto vertex_partition_range_lasts = graph_view.vertex_partition_range_lasts();
+
+  rmm::device_uvector<vertex_t> d_vertex_partition_range_lasts(vertex_partition_range_lasts.size(),
+                                                               handle.get_stream());
+
+  raft::update_device(d_vertex_partition_range_lasts.data(),
+                      vertex_partition_range_lasts.data(),
+                      vertex_partition_range_lasts.size(),
+                      handle.get_stream());
+
+  auto func = cugraph::detail::compute_gpu_id_from_int_edge_endpoints_t<vertex_t>{
+    raft::device_span<vertex_t const>(d_vertex_partition_range_lasts.data(),
+                                      d_vertex_partition_range_lasts.size()),
+    comm_size,
+    major_comm_size,
+    minor_comm_size};
+
+  rmm::device_uvector<vertex_t> cp_edgelist_srcs(edgelist_srcs.size(), handle.get_stream());
+  rmm::device_uvector<vertex_t> cp_edgelist_dsts(edgelist_srcs.size(), handle.get_stream());
+
+  thrust::copy(handle.get_thrust_policy(),
+               thrust::make_zip_iterator(edgelist_srcs.begin(), edgelist_srcs.begin()),
+               thrust::make_zip_iterator(edgelist_dsts.end(), edgelist_dsts.end()),
+               thrust::make_zip_iterator(cp_edgelist_srcs.begin(), cp_edgelist_dsts.begin()));
+
+  auto d_tx_counts = cugraph::groupby_and_count(
+    thrust::make_zip_iterator(edgelist_srcs.begin(), edgelist_dsts.begin()),
+    thrust::make_zip_iterator(edgelist_srcs.end(), edgelist_dsts.end()),
+    [func] __device__(auto val) { return func(val); },
+    comm_size,
+    std::numeric_limits<size_t>::max(),
+    handle.get_stream());
+
+  std::vector<size_t> h_tx_counts(d_tx_counts.size());
+
+  raft::update_host(
+    h_tx_counts.data(), d_tx_counts.data(), d_tx_counts.size(), handle.get_stream());
+
+  rmm::device_uvector<vertex_t> srcs(0, handle.get_stream());
+  rmm::device_uvector<vertex_t> dsts(0, handle.get_stream());
+  std::vector<size_t> rx_counts{};
+
+  std::tie(srcs, rx_counts) =
+    shuffle_values(handle.get_comms(), edgelist_srcs.begin(), h_tx_counts, handle.get_stream());
+
+  std::tie(dsts, std::ignore) =
+    shuffle_values(handle.get_comms(), edgelist_dsts.begin(), h_tx_counts, handle.get_stream());
+
+  // rmm::device_uvector<bool> edge_exists(0, handle.get_stream());
+
+  auto edge_exists =
+    graph_view.has_edge(handle,
+                        raft::device_span<vertex_t const>(srcs.data(), srcs.size()),
+                        raft::device_span<vertex_t const>(dsts.data(), dsts.size()));
+
+  // Send the result back
+  std::tie(edge_exists, std::ignore) =
+    shuffle_values(handle.get_comms(), edge_exists.begin(), rx_counts, handle.get_stream());
+
+  // The 'edge_exists' array is ordered based on 'cp_edgelist_srcs' where the edges where group,
+  // hoever it needs to match 'edgelist_srcs', hence re-order 'edge_exists' accordingly.
```

Comment:
Is this true? I guess `edge_exists` is ordered based on `edgelist_srcs` not `cp_edgelist_srcs`.

Reply:
Yes other way around. It is `cp_edgelist_srcs` that is supposed to be groupby_and_count. The documentation is correct but made a mistake in the group_by_and_count call.  I don't want to mess up original edges in the triangles when grouping that's why I make a copy.